### PR TITLE
Handle typeahead selections and free text in typeahead filters.

### PR DESCRIPTION
**Pull request by jmcarp**
Requires https://github.com/18F/fec-style/pull/231, https://github.com/18F/openFEC/pull/1462

cc @noahmanger @LindsayYoung @jenniferthibault

### CODE REVIEW

**Comment by jmcarp** [comment]
This is part of the consolidated typeahead feature. Assigned to @ccostino for review.

File: package.json [hunk 1]
```diff
@@ -64,7 +64,7 @@
     "d3": "3.5.5",
     "datatables.net": "1.10.10",
     "datatables.net-responsive": "2.0.1",
-    "fec-style": "1.4.1",
+    "fec-style": "git://github.com/jmcarp/fec-style.git#consolidate-typeahead",
```
Comment (ccostino) [inline package.json]: Just noting that this will have to change once the updates are applied to the `fec-style` repo.

**Comment by ccostino** [comment]
The `fec-style` module has been updated to `1.6.0` and is published.  This PR should be ready to go once the change is made.

**Comment by noahmanger** [comment]
The build is failing saying it can't find underscore: https://travis-ci.org/18F/openFEC-web-app/jobs/111576057#L730

I remember this happening, but not sure I learned why it happens?

**Pull request merged**